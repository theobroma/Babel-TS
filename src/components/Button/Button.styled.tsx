import { default as styled } from 'styled-components';

const StyledButton = styled.button`
  background: white;
  border: 1px solid black;
  border-radius: 4px;
  padding: 10px;
`;

export default StyledButton;
