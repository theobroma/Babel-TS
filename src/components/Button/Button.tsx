import * as React from 'react';

import { ButtonProps } from '../../declerations/Button.d';
import StyledButton from './Button.styled';

const Button: React.FunctionComponent<ButtonProps> = (props: ButtonProps) => (
  <StyledButton onClick={props.onClick}>
    {props.label}
  </StyledButton>
);

Button.defaultProps = {
  label: 'Hello World',
  onClick: (e) => {}
};

export default Button;