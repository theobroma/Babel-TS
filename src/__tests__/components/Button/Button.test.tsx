import * as React from 'react';
import {shallow} from 'enzyme';

import Button from '../../../components/Button/Button';

describe('<Button/>', () => {
  test('Should render', () => {
    const wrapper = shallow(<Button/>);
    expect(wrapper).toMatchSnapshot();
  })
})