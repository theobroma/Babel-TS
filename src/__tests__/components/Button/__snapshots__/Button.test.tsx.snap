// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should render 1`] = `
<button>
  test
</button>
`;
